lib/valueflow.cpp

```diff
@@ -2937,6 +2937,23 @@ static bool valueFlowForwardVariable(Token* const startToken,
                 return false;
         }
 
+        // Member function call
+        else if (!var->isLocal() && tok2->function() && !tok2->function()->isConst() && Token::Match(tok2, "%name% (") && exprDependsOnThis(tok2)) {
+            if (tok2->function()->functionScope) {
+                const bool changed = isVariableChanged(
+                                   tok2->function()->functionScope->bodyStart,
+                                   tok2->function()->functionScope->bodyEnd, varid,
+                                   var->isGlobal(), settings, tokenlist->isCPP());
+                if (changed) {
+                    if (settings->debugwarnings)
+                        bailout(tokenlist, errorLogger, tok2, "Variable was modified by member function call.");
+                    return false;
+                } 
```

Comment:
We should probably check that the function doesn't call another non-const member function. I dont think `isVariableChanged` checks for that.